Support ObjectId in MongoDB operators

melohagan opened the pull request:
## Description
Fix to recognise object ids inside MongoDB operators such as `$eq`

Addresses: 
- https://github.com/Budibase/budibase/issues/6778
- https://github.com/Budibase/budibase/issues/7079 
- https://github.com/Budibase/budibase/discussions/7056

Rory-Powell reviewed (approved):
Looks good mel. It would be nice to see a test as well where we can supply some variations of json queries and assert that mongo is called with the right data

melohagan commented:
@Rory-Powell I realised that the regex wasn't handling scenarios where there were braces `{}` within strings (double quotes). 
I removed the regex and did it programmatically because it was starting to get too complicated for my regex skills. 
Added a unit test to cover this scenario as well!

Rory-Powell reviewed (approved):
LGTM 👍
